src/System.Management.Automation/utils/PathUtils.cs

```diff
@@ -515,5 +542,104 @@ internal static Encoding Convert(Cmdlet cmdlet, string encoding)
 
             return null;
         }
+
+        // We need a way to provide an encoding and also
+        // a way to notify that the user wants "byte" encoding
+        // which is just a stream
+        internal class ByteEncoding : System.Text.Encoding
+        {
+            // the encoding for this is not bigendian and not BOM
+            public Encoding ActualEncoding = new UnicodeEncoding(false, false);
+            public override unsafe int GetByteCount(char* c, int i1) { throw new NotImplementedException(); }
+            public override int GetByteCount(char[] c, int i1, int i2) { throw new NotImplementedException(); }
```

Comment:
I believe it's not hard for someone to get an instance of this encoding, so maybe we should be implementing all of these methods, assuming it's straightforward to do so (and mostly it seems like it is.)

Reply:
done